src/transport_portals4.h

```diff
@@ -1314,4 +1314,79 @@ void shmem_transport_syncmem(void)
     PtlAtomicSync();
 }
 
+#ifdef ENABLE_THREADS
+#define SHMEM_TRANSPORT_PTL4_CNTR_LOCK(ctx) SHMEM_MUTEX_LOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_UNLOCK(ctx) SHMEM_MUTEX_UNLOCK(ctx)
+#define SHMEM_TRANSPORT_PTL4_CNTR_READ(cntr) *(cntr) 
+#else 
+#define SHMEM_TRANSPORT_PTL4_CNTR_LOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_UNLOCK(ctx) 
+#define SHMEM_TRANSPORT_PTL4_CNTR_READ(cntr) shmem_internal_atomic_read(cntr) 
+#endif
+
+static inline
+uint64_t shmem_transport_get_pending_put_cntr(shmem_transport_ctx_t *ctx)
+{
+    uint64_t cnt = 0;
+    SHMEM_TRANSPORT_PTL4_CNTR_LOCK(shmem_internal_mutex_ptl4_ctx);
+    cnt = SHMEM_TRANSPORT_PTL4_CNTR_READ(&shmem_transport_portals4_pending_put_event_cntr);
+    cnt += SHMEM_TRANSPORT_PTL4_CNTR_READ(&ctx->pending_put_cntr);
+    SHMEM_TRANSPORT_PTL4_CNTR_UNLOCK(shmem_internal_mutex_ptl4_ctx);
+    return cnt;
+}
+
+static inline
+uint64_t shmem_transport_get_pending_get_cntr(shmem_transport_ctx_t *ctx)
+{
+    uint64_t cnt = 0;
+    SHMEM_TRANSPORT_PTL4_CNTR_LOCK(shmem_internal_mutex_ptl4_ctx);
```

Comment:
No need to lock around calls to PtlCTGet; it's already thread safe.  This lock object is used to guard the context tracking array, so it won't help here.